core/conn/unixodbc/odbc/odbcclient/unixcli/DrvrManager/drvrmanager.cpp

```diff
@@ -3132,42 +3132,38 @@ SQLRETURN  SQL_API SQLGetTypeInfo_common(SQLHSTMT StatementHandle,
 		RETURNCODE (pstmt, SQL_ERROR);
 		LEAVE_STMT (pstmt, SQL_ERROR);
 	}
-	SQLINTEGER ODBCAppVersion = pstmt->getODBCAppVersion();
 	SQLSMALLINT convDataType = DataType;
```

Comment:
Isn't DriverManager supposed to do the mapping based on ODBC app version at runtime?  I am thinking that the old code is doing the right thing. A unit test using ODBC 2.0 and ODBC 3.0 application would help to determine which way to go.